cookbooks/aws-parallelcluster-slurm/recipes/install_pmix.rb

```diff
@@ -26,15 +26,8 @@
   mode '0644'
   retries 3
   retry_delay 5
-  not_if { ::File.exist?(pmix_tarball) }
-end
-
-ruby_block "Validate PMIx Tarball Checksum" do
-  block do
-    require 'digest'
-    checksum = Digest::SHA256.file(pmix_tarball).hexdigest
-    raise "Downloaded Tarball Checksum #{checksum} does not match expected checksum #{pmix_sha256}" if checksum != pmix_sha256
-  end
+  checksum pmix_sha256
```

Comment:
From [chef-docs](https://docs.chef.io/resources/remote_file/) it looks like checksum is used to prevent a file from being re-downloaded, when the local file matches the checksum.

All the checksums present in our cookbook have a different goal which is to verify the integrity of the downloaded artifact and prevent the potential download of compromised software. Does the new checksum mechanism achieves the same result?

Reply:
According to this comment in the code it seems so

https://github.com/chef/chef/blob/main/lib/chef/resource/remote_file.rb#L45-L63

If the file doesn't exist it is downloaded and the checksum, validated.
If the file exists and the checksum match it is kept and a new download avoided.

Reply:
Chef documentation is not clear: they don't say the checksum is used *only* to that end. In fact, the second example on the page you linked says "Download a file with a checksum to validate". This is why we tested the behavior [in a previous PR](https://github.com/aws/aws-parallelcluster-cookbook/pull/1772#discussion_r1108657687) before changing the code.